lm-cli/lm_cli/requests.py

```diff
@@ -0,0 +1,191 @@
+"""
+Provide utilities for making requests against the License Manager API.
+"""
+from typing import Any, Dict, Optional, Type, TypeVar, Union
+
+import httpx
+import pydantic
+from loguru import logger
+
+from lm_cli.exceptions import Abort
+from lm_cli.text_tools import dedent, unwrap
+
+
+def _deserialize_request_model(
+    request_model: pydantic.BaseModel,
+    request_kwargs: Dict[str, Any],
+    abort_message: str,
+    abort_subject: str,
+):
+    """
+    Deserialize a pydantic model instance into request_kwargs for an httpx client request in place.
+    """
+    Abort.require_condition(
+        all(
+            [
+                "data" not in request_kwargs,
+                "json" not in request_kwargs,
+                "content" not in request_kwargs,
+            ]
+        ),
+        unwrap(
+            f"""
+            {abort_message}:
+            Request was incorrectly structured.
+            """
+        ),
+        raise_kwargs=dict(
+            subject=abort_subject,
+            support=True,
+            log_message=unwrap(
+                """
+                When using `request_model`, you may not pass
+                `data`, `json`, or `content` in the `request_kwargs`
+                """
+            ),
+        ),
+    )
+    try:
+        request_kwargs["content"] = request_model.json()
+        request_kwargs["headers"] = {"Content-Type": "application/json"}
+    except Exception as err:
+        raise Abort(
+            unwrap(
+                f"""
+                {abort_message}:
+                Request data could not be deserialized for http request.
+                """
+            ),
+            subject=abort_subject,
+            support=True,
+            log_message=unwrap(
+                f"""
+                Could not deserialize instance of {request_model.__class__}:
+                {request_model}
+                """
+            ),
+            original_error=err,
+        )
+
+
+ResponseModel = TypeVar("ResponseModel", bound=pydantic.BaseModel)
+
+
+def make_request(
+    client: httpx.Client,
+    url_path: str,
+    method: str,
+    *,
+    expected_status: Optional[int] = None,
+    expect_response: bool = True,
+    abort_message: str = "There was an error communicating with the API",
```

Comment:
```suggestion
    abort_message: str = "There was an error communicating with the API.",
```

Reply:
The `abort_message` is concatenated with more information, so it shouldn't have a trailing period.